src/widget/text_box.rs

```diff
@@ -248,71 +256,101 @@ fn closest_idx<C: CharacterCache>(glyph_cache: &GlyphCache<C>,
-            Up => {
-                prev.cursor.anchor = Anchor::None;
-                Captured(prev)
-            },
-        },
-
-        Interaction::Uncaptured(prev) => match mouse.left.position {
-            Down => match over_elem {
-                Elem::Nill => Uncaptured(Normal),
-                Elem::Rect => match prev {
-                    Normal => prev_interaction,
-                    Highlighted => Captured(View {
-                         cursor: Cursor::from_index(0),
-                         offset: 0.0,
-                    })
-                },
-                Elem::Char(idx) =>  match prev {
-                    Normal => prev_interaction,
-                    Highlighted => Captured(View {
-                        cursor: Cursor::from_index(idx),
-                        offset: 0.0,
-                    })
-                },
-            },
-            Up => match over_elem {
-                Elem::Nill => Uncaptured(Normal),
-                Elem::Char(_) | Elem::Rect => match prev {
-                    Normal => Uncaptured(Highlighted),
-                    Highlighted => prev_interaction,
-                },
-            },
-        },
+                }
+            }
+        }
+
+        Interaction::Uncaptured(prev) => {
+            match mouse.left.position {
+                Down => {
+                    match over_elem {
+                        Elem::Nill => Uncaptured(Normal),
+                        Elem::Rect => {
+                            match prev {
+                                Normal => prev_interaction,
+                                Highlighted => {
+                                    Captured(View {
+                                        cursor: Cursor::from_index(0),
+                                        offset: 0.0,
+                                    })
+                                }
+                            }
+                        }
+                        Elem::Char(idx) => {
+                            match prev {
+                                Normal => prev_interaction,
+                                Highlighted => {
+                                    Captured(View {
+                                        cursor: Cursor::from_index(idx),
+                                        offset: 0.0,
+                                    })
+                                }
+                            }
+                        }
+                    }
+                }
+                Up => {
+                    match over_elem {
+                        Elem::Nill => Uncaptured(Normal),
+                        Elem::Char(_) | Elem::Rect => {
+                            match prev {
+                                Normal => Uncaptured(Highlighted),
+                                Highlighted => prev_interaction,
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
The extra braces on all of these branches is too much, I think I much prefer the original. I wonder why rustfmt always tries to nest match exprs in another set of braces, when they basically are a set of braces themselves?